homeassistant/components/ovo_energy/config_flow.py

```diff
@@ -66,7 +69,9 @@ async def async_step_reauth(self, user_input):
             client = OVOEnergy()
             try:
                 authenticated = await client.authenticate(
-                    self.username, user_input[CONF_PASSWORD]
+                    self.username,
+                    user_input[CONF_PASSWORD],
+                    user_input.get(ACCOUNT, None),
```

Comment:
Why re specify on reauth?

Reply:
probably because I wasnt sure on re-auth the underlying library it uses doesnt reset the account if not specified

Reply:
thanks will do, looking at other integrations Im not sure why the original code is even using user_input in this step, shouldnt it just be reading the config settings?

Reply:
Im getting confused over what to do here. My limited understanding of the reauth step is that HA uses it when the component fails its initial authorization for some reason. If this is the case if I remove the account value from the call the the OVOEnergy client "authenticate" call then the OVOEnergy client library will choose the wrong account OR are you saying this step should not be getting the value from the user_input but instead read it from the config, if so I was following the existing code which is also therefore not correct and Ill need to change. Apologies but Im new to HA so have a lot to learn about standards

Reply:
It should get it from config. It already grabs username from there.

Reply:
then I have no idea how to get this code to be what it needs to be, the current code does not get username from config it does gets its from user input as well (line 67ish):

```
        if user_input and user_input.get(CONF_USERNAME):
            self.username = user_input[CONF_USERNAME]
```

Reply:
Do it the same way. The initial call to re-auth will contain the original config entry data as user_input.

Reply:
ll do that then, but still cant see how getting it from user input a few lines earlier and setting self.account in the same method is any different than getting it from user input when calling the client.authenticate what am I missing here?

Reply:
The dialog the user get shown should not include account, thus the second time into that function the user data will not contain account.

Reply:
what happens if the reason the auth failed was because they entered an account id that is not valid? The underlying authenticate method has this code:

```
                    if account:
                        if account in self._account_ids:
                            self._account_id = account
                        else:
                            return False
```

Which as far as I can see means if I enter an account id that is not returned in the data from the provider then the authenticate method will fail and therefore the user might need to change the account id?

Reply:
That should have been handled by the standard flow. It should have validated the account. If its a new account, the user need to re-setup.

Reply:
Ok now I've worked out how to trigger async_step_reauth and can see how it works I believe these latest changes are what was being requested. Please can you confirm this is now as expected